common/CMakeLists.txt

```diff
@@ -0,0 +1,1125 @@
+        ${COMMON_SOURCE_DIR}/IO/AseParser.h
+        ${COMMON_SOURCE_DIR}/IO/File.h
+        ${COMMON_SOURCE_DIR}/IO/Reader.h
+        ${COMMON_SOURCE_DIR}/Renderer/GLVertex.h
+        ${COMMON_SOURCE_DIR}/Renderer/GLVertexAttributeType.h
+        ${COMMON_SOURCE_DIR}/Renderer/GLVertexType.h
+        ${COMMON_SOURCE_DIR}/StepIterator.h
+)
+
+set(CMAKE_AUTOMOC ON)
+
+add_library(common OBJECT ${COMMON_SOURCE} ${COMMON_HEADER})
+target_compile_features(common PRIVATE cxx_std_17)
+target_include_directories(common PUBLIC ${COMMON_SOURCE_DIR})
+target_link_libraries(common PUBLIC tinyxml2 vecmath optlite glew miniz freeimage freetype OpenGL::GL Qt5::Widgets)
+
+if(WIN32)
+    target_link_libraries(common PRIVATE stackwalker)
+endif()
+
+if(APPLE)
+    # Silence macOS OpenGL deprecation warnings
+    target_compile_definitions(common PUBLIC GL_SILENCE_DEPRECATION)
+endif()
+
+if(COMPILER_IS_CLANG)
+    target_compile_options(common PRIVATE -Wall -Wextra -Weverything -pedantic -Wno-format -Wno-variadic-macros -Wno-c99-extensions -Wno-padded -Wno-unused-parameter -Wno-global-constructors -Wno-exit-time-destructors -Wno-weak-vtables -Wno-weak-template-vtables -Wno-float-equal -Wno-used-but-marked-unused -Wno-format-nonliteral -Wno-missing-noreturn -Wno-unused-local-typedef -Wno-double-promotion -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-implicit-fallthrough -Wno-zero-as-null-pointer-constant -Wno-switch-enum)
+    target_compile_options(common PRIVATE "$<$<CONFIG:RELEASE>:-O3>")
+
+    # FIXME: Remove once we switch to Xcode 10
+    target_compile_options(common PRIVATE -Wno-missing-braces)
+
+    # FIXME: Suppress warnings in moc generated files:
+    target_compile_options(common PRIVATE -Wno-redundant-parens)
+elseif(COMPILER_IS_GNU)
+    target_compile_options(common PRIVATE -Wall -Wextra -pedantic -Wno-format -Wno-variadic-macros -Wno-padded -Wno-unused-parameter -Wno-float-equal -Wno-format-nonliteral -Wno-missing-noreturn -Wno-zero-as-null-pointer-constant -Wno-error=maybe-uninitialized)
+    target_compile_options(common PRIVATE "$<$<CONFIG:RELEASE>:-O3>")
+
+    # FIXME: enable -Wcpp once we found a workaround for glew / QOpenGLWindow problem, see RenderView.h
+    target_compile_options(common PRIVATE -Wno-cpp)
+elseif(COMPILER_IS_MSVC)
+    target_compile_definitions(common PRIVATE _CRT_SECURE_NO_DEPRECATE _CRT_NONSTDC_NO_DEPRECATE)
+    target_compile_options(common PRIVATE /W3 /EHsc /MP)
+    target_compile_options(common PRIVATE "$<$<CONFIG:RELEASE>:/Ox>")
+
+    # Generate debug symbols even for Release; we build a stripped pdb in Release mode, see TrenchBroomApp.cmake
+    target_compile_options(common PRIVATE "$<$<CONFIG:RELEASE>:/Zi>")
+
+    # Request 8MB stack on Windows.
+    # Default is 1MB and we need more for our current AABB builder on large maps (https://github.com/kduske/TrenchBroom/issues/2803)
+    set(TB_STACK_SIZE 8388608)
+    set_target_properties(common PROPERTIES LINK_FLAGS "/STACK:${TB_STACK_SIZE}")
```

Comment:
`/STACK` should only be used on .exe targets, is that what this does?

Just doing it on all .exe's in the project as we were before seems less error prone (e.g. this method would break if common was compiled as a static lib)

Reply:
This didn't work, so I reverted to the global CMAKE_EXE_LINKER_FLAGS setting